### ISSUE TITLE
<label> font-size larger than expected when .form-group-sm stacks

### ISSUE THREAD

**Issue by ddelella** [twbs/bootstrap#16938]
I am trying to create a responsive form.  The default form layout is horizontal.  When the user hits the small browser size it will put the form back into the standard form layout with label on top of the input field.  I have my default form group class set to "form-group form-group-sm".  This seems to display just fine in the horizontal view but when it changes to the standard form layout the font styles appear to be reset to the medium.

I have attached a working sample via JSFiddle.  In the example, if you slide the middle divider to simulate window size in the result window you will see the issue in the first text box field.  The second textbox is meant to simulate the medium form group size.  I don't know if this is by design but it causes inconsistent form layout between devices.

https://jsfiddle.net/ddelella/02vaLa9s/8/

**Comment by twbs-lmvtfy** [comment]
Hi @ddelella!

You appear to have posted a live example (https://fiddle.jshell.net/ddelella/02vaLa9s/8/show/light/), which is always a good first step. However, according to [the HTML5 validator](http://html5.validator.nu), **your example has some validation errors**, which might potentially be causing your issue:
* line 42, column 3 thru line 42, column 42: Bad value ````  ```` for attribute ```` action ```` on element ```` form ````: Must be non-empty.

You'll need to **fix these errors** and post a revised example before we can proceed further.
Thanks!

(*Please note that this is a [fully automated](https://github.com/cvrebert/lmvtfy) comment.*)

**Comment by kappasims** [comment]
I'm experiencing the same issue. I've attempted to fix the validation in the fiddle:

https://jsfiddle.net/02vaLa9s/9/

The issue is that the font size rule for form-group-sm control labels is wrapped in the following media query:
```css
@media (min-width: 768px) {
  .form-horizontal .form-group-sm .control-label {
    padding-top: 6px;
    font-size: 12px;
  }
}
```

My workaround is to add the innards of the query to my site styles:
```css
.form-horizontal .form-group-sm .control-label {
  padding-top: 6px;
  font-size: 12px;
}
```

**Comment by kappasims** [comment]
I'm experiencing the same issue. I've attempted to fix the validation in the fiddle:

https://jsfiddle.net/02vaLa9s/9/

The issue is that the font size rule for form-group-sm control labels is wrapped in the following media query:

@media (min-width: 768px) {
  .form-horizontal .form-group-sm .control-label {
    padding-top: 6px;
    font-size: 12px;
  }
}

My workaround is to add the innards of the query to my site styles:

.form-horizontal .form-group-sm .control-label {
  padding-top: 6px;
  font-size: 12px;
}

**Comment by mdo** [comment]
Punting as a won't fix given how late into v3's lifecycle we are. <3
